packages/website/src/components/Playground.tsx

```diff
@@ -115,46 +105,43 @@ function Playground(): React.JSX.Element {
             active={activeTab}
             change={setTab}
             showVisualEditor={activeTab !== 'code'}
-            showModal={onVisualEditor}
+            showModal={useCallback(tab => {
+              if (tab === 'tsconfig') {
+                setVisualTSConfig(val => !val);
+              } else if (tab === 'eslintrc') {
+                setVisualEslintRc(val => !val);
+              }
+            }, [])}
           />
-          {(activeVisualEditor === 'eslintrc' && (
-            <ConfigEslint
+          {ActiveVisualEditor && (
+            <ActiveVisualEditor
               className={styles.tabCode}
               ruleOptions={ruleNames}
-              config={state.eslintrc}
+              config={state[activeTab]}
               onChange={setState}
             />
-          )) ||
-            (activeVisualEditor === 'tsconfig' && (
-              <ConfigTypeScript
-                className={styles.tabCode}
-                config={state.tsconfig}
-                onChange={setState}
-              />
-            ))}
+          )}
           <div
             key="monacoEditor"
             className={clsx(
               styles.tabCode,
-              !!activeVisualEditor && styles.hidden,
+              ActiveVisualEditor && styles.hidden,
             )}
           >
             <EditorEmbed />
           </div>
           <LoadingEditor
-            ts={state.ts}
-            fileType={state.fileType}
+            {...state}
             activeTab={activeTab}
-            code={state.code}
-            tsconfig={state.tsconfig}
-            eslintrc={state.eslintrc}
-            sourceType={state.sourceType}
-            showAST={state.showAST}
             onASTChange={setAstModel}
             onMarkersChange={setMarkers}
             selectedRange={selectedRange}
             onChange={setState}
-            onLoaded={onLoaded}
+            onLoaded={useCallback((ruleNames, tsVersions) => {
```

Comment:
[Refactor] Err, let's not put `use*` hooks inside components. The [React rules of hooks](https://react.dev/reference/rules/rules-of-hooks) get very hard to keep track of when hooks are called inline.